### ISSUE TITLE
Accessibility problem in Menu component - @mantine/core

### ISSUE THREAD

**Issue by jonathanhuynh70** [mantinedev/mantine#8594]
### Dependencies check up

- [x] I have verified that I use latest version of all @mantine/* packages

### What version of @mantine/* packages do you have in package.json?

8.3.11

### What package has an issue?

@mantine/core

### What framework do you use?

Vite

### In which browsers you can reproduce the issue?

Not applicable – issue is not related to the browser

### Describe the bug

The Menu Target component has a attribute called aria-controls. This attribute has the id ( 'aria-controls': "dropdown-id") of the dropdown that it controls.
The problem is when the dropdown is closed the aria-controls attribute is still existing on the Menu Target while the dropdown is not even visible in the DOM, so it points to nothing.

Currently, I am getting an accessibility warning in storybook:

Unable to determine if aria-controls referenced ID exists on the page while using aria-haspopup: aria-controls="mantine-xxxxxxxxx-dropdown"

This is very similar to this issue: https://github.com/mantinedev/mantine/issues/7085. I believe it would be a similarly simple fix. 

### Possible fix

I think there is a possible fix in https://github.com/mantinedev/mantine/blob/master/packages/%40mantine/core/src/components/Popover/PopoverTarget/PopoverTarget.tsx#L51.  
**Proposed change:** 'aria-controls': ctx.opened ? ctx.getDropdownId() : undefined. 

### Self-service

- [x] I would be willing to implement a fix for this issue

**Comment by rtivital** [comment]
You are welcome to submit a PR with a fix

**Comment by J-P-Robin** [comment]
I'm having the same issue. I had to set `keepMounted` to `true`. Although the keyboard navigation (arrows) stops working in this case. The component does not catch the keyboard event when the dropdown is open. The quickfix was to manage by hand the first focus.

This second issue may not be related to this one, but thought it could be helpful to know.
